### PR TITLE
Added a Pulumi example with Wrangler and a dynamic provider

### PR DESCRIPTION
**Pull request by cf-jas**
### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

**Comment by cf-jas** [comment]
reviewing

**Comment by cf-jas** [comment]
Reviewed

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    create: `npx wrangler d1 migrations apply mydb --remote`,
+    triggers:[
+        hashDirectory(`${d1Dir}migrations`)
+    ]
+}, { dependsOn: [myD1Database] });
+
+// Run 'wrangler' command
+//    https://www.pulumi.com/registry/packages/command/api-docs/local/command/
+//--------------------------------------------------------------------------------------------------------------------
+let workerDir="../../worker-test/";
```
Comment (craigsdennis) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Why the let here?

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+In this example besides a Zone we also create a Worker, a Zero Trust Applications, Zero Trust Policies, a D1 database among others.
+
+Check also how to:
+
+- create Workers by calling Wrangler directly, instead of using the Cloudflare Pulumi provider supported Worker resource. The advantage of this method is that you can use it for any other deployment related task like executing D1 migrations as seen in the example below. When running the IaC script, Wrangler will create or update Workers and D1 (also performing migrations). Observe that D1 migrations state in Pulumi is only changed when the migrations directory hash changes triggering that Pulumi action.
+- create a dynamic resource provider for a resource not supported. In this example we demonstrate how to do it for Vectorize.
+
+
+```
```
Comment (craigsdennis) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: I think there is a new GitHub Component where you could point to this repo. @KianNH is that ready?

Comment (craigsdennis) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: If not, let's mark this javascript

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+## Accessing Pulumi exports
+
+After running your Pulumi script with "pulumi up" you will have your resources created or updated. 
+
+The script above also exports outputs that you can access from other tools that you want or need to integrate into your deployment pipeline.
+
+For this, you can use: 
+
+```
```
Comment (craigsdennis) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Let's mark this bash

If you don't put the $ I think you can copy paste it

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+const myD1Database = new cloudflare.D1Database("myD1Database", {
+    accountId: accountId,
+    name: "mydb",
+}, { provider: cloudflareProvider });
+
+// Deploy Changes to D1 Schema
+//  Cloudflare Wrangler stores list of migrations in the D1 database.
+//  To check which migrations were ran go to the Cloudflare dashboard and run "select * from d1_migrations" on the console of the D1 database
+//--------------------------------------------------------------------------------------------------------------------
+let d1Dir="../../mydb/";
```
Comment (craigsdennis) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Let here as well?

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${props.apiToken}`
+            }
+        };        
+        // Make an API call to delete the resource
+        await axios.delete(url, options);
+    },
+
+    async update(id, oldInputs, newInputs) {
+        // Vectorize once created does not allow updates
```
Comment (craigsdennis) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Is there a practice of throwing errors here?

**Review by craigsdennis — APPROVED** [review]
Looks good, just a few minor nits, and an ask that if there is a repo available we use that (and link to it)

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+## Accessing Pulumi exports
+
+After running your Pulumi script with "pulumi up" you will have your resources created or updated. 
+
+The script above also exports outputs that you can access from other tools that you want or need to integrate into your deployment pipeline.
+
+For this, you can use: 
+
+```
+  $ pulumi stack output myServiceTokenClientSecret --show-secrets
```
Comment (pedrosousa) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]:
```suggestion
```bash
pulumi stack output myServiceTokenClientSecret --show-secrets
```

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 2 above)
Comment (pedrosousa) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Marked as JavaScript.

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 4 above)
Comment (pedrosousa) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Changed to `const`.

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 1 above)
Comment (pedrosousa) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Changed to `const`.

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+				"Content-Type": "application/json",
+				Authorization: `Bearer ${props.apiToken}`,
+			},
+		};
+		// Make an API call to delete the resource
+		await axios.delete(url, options);
+	},
+
+	async update(id, oldInputs, newInputs) {
+		// Vectorize once created does not allow updates
```
Comment (pedrosousa) [inline src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx:292]: Recovering the comment from Dennis about this piece of code:
> Is there a practice of throwing errors here?

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 5 above)
Comment (pedrosousa) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: This was not addressed in my (PCX) changes (I prettified the code and now the comment is outdated). Still not sure if we want to throw some error here, or if we can merge as is.

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 1 above)
Comment (cf-jas) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Unfortunately, there isn't a git repo with this example.

File: src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx (hunk 5 above)
Comment (cf-jas) [reply src/content/docs/pulumi/tutorial/dynamic-provider-and-wrangler.mdx]: Merge seems a good option, and the example can be refined in future iterations if needed.

**Review by pedrosousa — APPROVED** [review]

**Pull request merged**